### PR TITLE
Add reference to hidden_pii to cutom events

### PR DESCRIPTION
**Pull request by ericaporter**
This PR includes an update to the Readme and a event test to cover hidden_data being included in in a custom event

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ it might be necessary to add a primary key to the table and to update the releva
 
 ## Custom events
 
-If you wish to send custom analytics event, create a file `config/analytics_custom_events.yml` containing an array of your custom events types under a `shared` key like:
+If you wish to send custom analytics event, for example if you have data about emails sent, server side validation errors, api query data or data relating to searches performed, create a file `config/analytics_custom_events.yml` containing an array of your custom events types under a `shared` key like:
```
Comment (stevenleggdfe) [inline README.md:259]: Perhaps capitalise API?

File: README.md [hunk 2]
```diff
@@ -275,6 +275,20 @@ event = DfE::Analytics::Event.new
   .with_data(some: 'custom details about event')
 ```
 
+If you need to include hidden_pii, you can use the `hidden_data` key which will allow all fields lsited to be sent separately to BigQuery where they will be hidden.
```
Comment (stevenleggdfe) [inline README.md:278]: Typo: listed not lsited

File: README.md [hunk 3]
```diff
@@ -275,6 +275,20 @@ event = DfE::Analytics::Event.new
...
+
+```ruby
+event = DfE::Analytics::Event.new
+  .with_type(:some_custom_event)
+  .with_user(current_user)
+  .with_request_details(request)
+  .with_namespace('some_namespace')
+  .with_data(
+    data: { some: 'custom details about event' },
```
Comment (stevenleggdfe) [inline README.md:287]: Would it be worth including more than one field in data and hidden_data in the example so it's super clear how to achieve this?

**Review by stevenleggdfe — CHANGES_REQUESTED** [review]
See comments

File: README.md (hunk 1 above)
Comment (ericaporter) [reply README.md]: Now updated

File: README.md (hunk 2 above)
Comment (ericaporter) [reply README.md]: Now updated

File: README.md (hunk 3 above)
Comment (ericaporter) [reply README.md]: Now updated

**Review by stevenleggdfe — APPROVED** [review]

**Review by asatwal — APPROVED** [review]
👍  LGTM

**Pull request merged**